// alipay-serverless-sdk/src/aliyun/shortMessage/client.ts
// This file is auto-generated, don't edit it
import CloudKernel from 'mpserverless-sdk-core';
import { Readable } from 'stream';
import * as $tea from '@alicloud/tea-typescript';

export class SendSmsResponse extends $tea.Model {
  requestId: string;
  code: string;
  message: string;
  bizId: string;
  static names(): { [key: string]: string } {
    return {
      requestId: 'RequestId',
      code: 'Code',
      message: 'Message',
      bizId: 'BizId',
    };
  }

  static types(): { [key: string]: any } {
    return {
      requestId: 'string',
      code: 'string',
      message: 'string',
      bizId: 'string',
    };
  }

  constructor(map?: { [key: string]: any }) {
    super(map);
  }
}

export class SendBatchSmsResponse extends $tea.Model {
  requestId: string;
  code: string;
  message: string;
  bizId: string;
  static names(): { [key: string]: string } {
    return {
      requestId: 'RequestId',
      code: 'Code',
      message: 'Message',
      bizId: 'BizId',
    };
  }

  static types(): { [key: string]: any } {
    return {
      requestId: 'string',
      code: 'string',
      message: 'string',
      bizId: 'string',
    };
  }

  constructor(map?: { [key: string]: any }) {
    super(map);
  }
}

export class QuerySmsResponse extends $tea.Model {
  requestId: string;
  code: string;
  message: string;
  smsSendDetailDTOs: SendDetailInfo;
  totalCount: number;
  static names(): { [key: string]: string } {
    return {
      requestId: 'RequestId',
      code: 'Code',
      message: 'Message',
      smsSendDetailDTOs: 'SmsSendDetailDTOs',
      totalCount: 'TotalCount',
    };
  }

  static types(): { [key: string]: any } {
    return {
      requestId: 'string',
      code: 'string',
      message: 'string',
      smsSendDetailDTOs: SendDetailInfo,
      totalCount: 'number',
    };
  }

  constructor(map?: { [key: string]: any }) {
    super(map);
  }
}

export class SendDetailInfo extends $tea.Model {
  smsSendDetailDTO: {[key: string]: any}[];
  static names(): { [key: string]: string } {
    return {
      smsSendDetailDTO: 'SmsSendDetailDTO',
    };
  }

  static types(): { [key: string]: any } {
    return {
      smsSendDetailDTO: { 'type': 'array', 'itemType': 'object' },
    };
  }

  constructor(map?: { [key: string]: any }) {
    super(map);
  }
}


export default class Client {
  _kernel: CloudKernel;

  constructor(_kernel: CloudKernel) {
    this._kernel = _kernel;
  }


  async send(phoneNumbers: string, signName: string, templateCode: string, templateParam: string): Promise<SendSmsResponse> {
    let params : {[key: string ]: any} = {
      PhoneNumbers: phoneNumbers,
      SignName: signName,
      TemplateCode: templateCode,
      TemplateParam: templateParam,
    };
    return $tea.cast<SendSmsResponse>(await this._kernel.invokePre("alicloud-dysms", "SendSms", params), new SendSmsResponse({}));
  }

  async batchSend(phoneNumberJson: {[key: string]: any}, signNameJson: {[key: string]: any}, templateCode: string, templateParamJson: string): Promise<SendBatchSmsResponse> {
    let params : {[key: string ]: any} = {
      PhoneNumberJson: phoneNumberJson,
      SignNameJson: signNameJson,
      TemplateCode: templateCode,
      TemplateParamJson: templateParamJson,
    };
    return $tea.cast<SendBatchSmsResponse>(await this._kernel.invokePre("alicloud-dysms", "SendBatchSms", params), new SendBatchSmsResponse({}));
  }

  async query(phoneNumber: string, sendDate: string, pageSize: number, currentPage: number): Promise<QuerySmsResponse> {
    let params : {[key: string ]: any} = {
      PhoneNumber: phoneNumber,
      SendDate: sendDate,
      PageSize: pageSize,
      CurrentPage: currentPage,
    };
    return $tea.cast<QuerySmsResponse>(await this._kernel.invokePre("alicloud-dysms", "QuerySendDetails", params), new QuerySmsResponse({}));
  }

  async preciseQuery(phoneNumber: string, sendDate: string, pageSize: number, currentPage: number, bizId: string): Promise<QuerySmsResponse> {
    let params : {[key: string ]: any} = {
      PhoneNumber: phoneNumber,
      SendDate: sendDate,
      PageSize: pageSize,
      CurrentPage: currentPage,
      BizId: bizId,
    };
    return $tea.cast<QuerySmsResponse>(await this._kernel.invokePre("alicloud-dysms", "QuerySendDetails", params), new QuerySmsResponse({}));
  }

}
