// alipay-serverless-sdk/src/marketing/templateMessage/client.ts
// This file is auto-generated, don't edit it
import CloudKernel from 'mpserverless-sdk-core';
import { Readable } from 'stream';
import * as $tea from '@alicloud/tea-typescript';

export class AlipayOpenAppMiniTemplatemessageSendResponse extends $tea.Model {
  code: string;
  msg: string;
  subCode: string;
  subMsg: string;
  static names(): { [key: string]: string } {
    return {
      code: 'code',
      msg: 'msg',
      subCode: 'subCode',
      subMsg: 'subMsg',
    };
  }

  static types(): { [key: string]: any } {
    return {
      code: 'string',
      msg: 'string',
      subCode: 'string',
      subMsg: 'string',
    };
  }

  constructor(map?: { [key: string]: any }) {
    super(map);
  }
}


export default class Client {
  _kernel: CloudKernel;

  constructor(_kernel: CloudKernel) {
    this._kernel = _kernel;
  }


  async send(toUserId: string, formId: string, userTemplateId: string, page: string, data: string): Promise<AlipayOpenAppMiniTemplatemessageSendResponse> {
    let params : {[key: string ]: any} = {
      to_user_id: toUserId,
      form_id: formId,
      user_template_id: userTemplateId,
      page: page,
      data: data,
    };
    return $tea.cast<AlipayOpenAppMiniTemplatemessageSendResponse>(await this._kernel.invoke("alipay.open.app.mini.templatemessage.send", params), new AlipayOpenAppMiniTemplatemessageSendResponse({}));
  }

}
