// alipay-serverless-sdk/src/base/qrcode/client.ts
// This file is auto-generated, don't edit it
import CloudKernel from 'mpserverless-sdk-core';
import { Readable } from 'stream';
import * as $tea from '@alicloud/tea-typescript';

export class AlipayOpenAppQrcodeCreateResponse extends $tea.Model {
  code: string;
  msg: string;
  subCode: string;
  subMsg: string;
  qrCodeUrl: string;
  static names(): { [key: string]: string } {
    return {
      code: 'code',
      msg: 'msg',
      subCode: 'subCode',
      subMsg: 'subMsg',
      qrCodeUrl: 'qrCodeUrl',
    };
  }

  static types(): { [key: string]: any } {
    return {
      code: 'string',
      msg: 'string',
      subCode: 'string',
      subMsg: 'string',
      qrCodeUrl: 'string',
    };
  }

  constructor(map?: { [key: string]: any }) {
    super(map);
  }
}


export default class Client {
  _kernel: CloudKernel;

  constructor(_kernel: CloudKernel) {
    this._kernel = _kernel;
  }


  async create(urlParam: string, queryParam: string, describe: string): Promise<AlipayOpenAppQrcodeCreateResponse> {
    let params : {[key: string ]: any} = {
      url_param: urlParam,
      query_param: queryParam,
      describe: describe,
    };
    return $tea.cast<AlipayOpenAppQrcodeCreateResponse>(await this._kernel.invoke("alipay.open.app.qrcode.create", params), new AlipayOpenAppQrcodeCreateResponse({}));
  }

}
