// @alicloud/mpserverless-core/dist/umd/transport/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constant_1 = require("../constant");
const codec_1 = require("../codec");
class HTTPTransport {
    constructor(endpoint) {
        this.endpoint = endpoint;
        this.protocol = constant_1.TransportProtocol.HTTP;
    }
    getEncoder(prefix) {
        return new codec_1.HTTPRequestEncoder(this.endpoint, prefix);
    }
    setAppId(appId) {
        this.appId = appId;
    }
    setAppSecret(privateKey) {
        this.appSecret = privateKey;
        return this;
    }
    setUA(ua) {
        this.ua = ua;
        return this;
    }
    setLogger(logger) {
        this.logger = logger;
    }
    setTimeout(timeout = '5s') {
        if (typeof timeout === 'string') {
            if (timeout.indexOf('ms') >= 0) {
                this.timeout = parseInt(timeout, 10);
                return;
            }
            if (timeout.indexOf('s') >= 0) {
                this.timeout = parseInt(timeout, 10) * 1000;
                return;
            }
        }
        else if (typeof timeout === 'number') {
            this.timeout = timeout;
            return;
        }
        this.timeout = 5000;
    }
    get timeoutOption() {
        return this.timeout;
    }
    setSpaceId(spaceId) {
        this.spaceId = spaceId;
    }
    request(encoder) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return (new codec_1.HTTPResponseDecoder()).decode();
        });
    }
}
exports.HTTPTransport = HTTPTransport;
