// @alicloud/mpserverless-miniapp-core/dist/esm/codec.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "tslib", "@alicloud/mpserverless-core", "crypto-js/core", "crypto-js/hmac-md5", "./error"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var tslib_1 = require("tslib");
    var mpserverless_core_1 = require("@alicloud/mpserverless-core");
    var core_1 = tslib_1.__importDefault(require("crypto-js/core"));
    var hmac_md5_1 = tslib_1.__importDefault(require("crypto-js/hmac-md5"));
    var error_1 = require("./error");
    var MPHTTPRequestEncoder = (function (_super) {
        tslib_1.__extends(MPHTTPRequestEncoder, _super);
        function MPHTTPRequestEncoder(endpoint, spaceId) {
            var _this = _super.call(this, endpoint) || this;
            _this.spaceId = spaceId;
            _this.prefix = mpserverless_core_1.PREFIX.CLIENT;
            _this.serviceHeaders = {};
            _this.setBodyField({
                spaceId: spaceId,
            });
            return _this;
        }
        MPHTTPRequestEncoder.prototype.sign = function (clientSecret) {
            var _a = this.body, spaceId = _a.spaceId, method = _a.method, params = _a.params, token = _a.token, userId = _a.userId;
            var timestamp = Date.now();
            this.setBodyField({
                timestamp: timestamp,
            });
            var signString = '';
            var signObject = {
                spaceId: spaceId,
                timestamp: timestamp,
                method: method,
                params: JSON.stringify(params),
                token: token,
                userId: userId,
            };
            Object.keys(signObject).sort().forEach(function (key) {
                if (signObject[key]) {
                    signString = signString + "&" + key + "=" + signObject[key];
                }
            });
            signString = signString.slice(1);
            var sign = hmac_md5_1.default(signString, clientSecret).toString(core_1.default.enc.Hex);
            this.setServerlessHeaders({ sign: sign });
        };
        MPHTTPRequestEncoder.prototype.encodeAsHTTPRequestObject = function (additionalObject) {
            if (this.body.params) {
                this.body.params = JSON.stringify(this.body.params);
            }
            return tslib_1.__assign({ url: this.url, data: this.body, method: this.method, headers: this.headers, dataType: 'json' }, additionalObject);
        };
        MPHTTPRequestEncoder.prototype.clone = function () {
            var encoder = new MPHTTPRequestEncoder(this.endpoint, this.spaceId);
            encoder.setBodyField(this.body);
            encoder.setBaseHeaders(this.baseHeaders);
            encoder.setServerlessHeaders(this.serverlessHeaders);
            return encoder;
        };
        return MPHTTPRequestEncoder;
    }(mpserverless_core_1.HTTPRequestEncoder));
    exports.MPHTTPRequestEncoder = MPHTTPRequestEncoder;
    var MPHTTPResponseDecoder = (function (_super) {
        tslib_1.__extends(MPHTTPResponseDecoder, _super);
        function MPHTTPResponseDecoder() {
            var _this = _super !== null && _super.apply(this, arguments) || this;
            _this.ERROR_CODES = [11, 12, 13, 14, 19, 20];
            return _this;
        }
        MPHTTPResponseDecoder.prototype.setStatusAndBody = function (status, body) {
            _super.prototype.setStatusAndBody.call(this, status, body);
            if (this.ERROR_CODES.indexOf(status) >= 0) {
                this._error = error_1.bizError.InterfaceRequestError("\u9519\u8BEF\u7801 " + status.toString() + ", \u8BF7\u53C2\u8003\u5BF9\u5E94\u5F00\u653E\u5E73\u53F0\u6587\u6863\u6392\u67E5\u95EE\u9898");
            }
        };
        MPHTTPResponseDecoder.prototype.decode = function (res) {
            this.setHeaders(res.headers || {});
            var body = res.data || res.body;
            if (typeof body === 'string') {
                body = JSON.parse(body);
            }
            if (body && body.data) {
                if (Number(body.data.affectedDocs) === body.data.affectedDocs) {
                    body = Object.assign({}, body, tslib_1.__assign({}, body.data));
                }
                else if (Object.prototype.toString.call(body.data) === '[object Object]') {
                    body.result = Object.assign({}, body.data);
                }
                else if (Object.prototype.toString.call(body.data) === '[object Array]') {
                    body.result = body.data.slice(0);
                }
                else {
                    body.result = body.data;
                }
                delete body.data;
            }
            if (/^request:fail+/.test(res.errMsg)) {
                this.setErrorMessage(res.errMsg);
                return _super.prototype.decode.call(this);
            }
            var responseErrorCode = parseInt(res.error, 10);
            if (responseErrorCode) {
                this.setStatusAndBody(responseErrorCode, body);
                return _super.prototype.decode.call(this);
            }
            var responseErrorMessage = res.err;
            if (responseErrorMessage) {
                this.setErrorMessage(responseErrorMessage);
                return _super.prototype.decode.call(this);
            }
            if (res instanceof Error) {
                this.setErrorObject(res);
                return _super.prototype.decode.call(this);
            }
            if (body && typeof body.error === 'object') {
                this.setErrorObject(body.error);
                return _super.prototype.decode.call(this);
            }
            var responseStatusCode = parseInt(res.status || res.statusCode, 10);
            if (responseStatusCode) {
                this.setStatusAndBody(responseStatusCode, body);
                return _super.prototype.decode.call(this);
            }
            this.setStatusAndBody(200, res);
            return _super.prototype.decode.call(this);
        };
        return MPHTTPResponseDecoder;
    }(mpserverless_core_1.HTTPResponseDecoder));
    exports.MPHTTPResponseDecoder = MPHTTPResponseDecoder;
});
