// @alicloud/mpserverless-miniapp-core/dist/esm/mpserverless.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "tslib", "@alicloud/mpserverless-core", "@alicloud/mp-proxy-service", "@alicloud/mp-db-service", "@alicloud/mp-user-service", "@alicloud/mp-function-service", "./transport", "./file"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var tslib_1 = require("tslib");
    var mpserverless_core_1 = require("@alicloud/mpserverless-core");
    var mp_proxy_service_1 = require("@alicloud/mp-proxy-service");
    var mp_db_service_1 = require("@alicloud/mp-db-service");
    var mp_user_service_1 = require("@alicloud/mp-user-service");
    var mp_function_service_1 = require("@alicloud/mp-function-service");
    var transport_1 = require("./transport");
    var file_1 = require("./file");
    var MPServerlessMiniAppCore = (function (_super) {
        tslib_1.__extends(MPServerlessMiniAppCore, _super);
        function MPServerlessMiniAppCore(appGlobal, options) {
            var _this = _super.call(this, tslib_1.__assign(tslib_1.__assign({}, options), { httpTransport: transport_1.MPHTTPTransport, options: appGlobal.logger })) || this;
            mpserverless_core_1.assert(options.clientSecret, 'clientSecret is required');
            mpserverless_core_1.assert(options.appId, 'appId is required');
            _this.user = new mp_user_service_1.UserService(_this.transport, {
                getAuthCode: _this.transport.wrapApi(appGlobal.getAuthCode),
                request: _this.transport.wrapApi(appGlobal.request),
                appId: options.appId,
                appSecret: options.clientSecret,
                ua: _this.ua,
            });
            _this.transport.setUserService(_this.user);
            _this.transport.setRequest(_this.transport.wrapApi(appGlobal.request));
            _this.db = new mp_db_service_1.DbService(_this.transport);
            _this.file = new file_1.HighFileSevice(_this.transport, {
                uploadFile: _this.transport.wrapApi(appGlobal.uploadFile),
                getFileUploadInfo: _this.transport.wrapApi(appGlobal.getFileUploadInfo),
            });
            _this.function = new mp_function_service_1.FunctionService(_this.transport);
            _this.network = new mp_proxy_service_1.ProxyService(_this.transport);
            return _this;
        }
        Object.defineProperty(MPServerlessMiniAppCore.prototype, "version", {
            get: function () {
                return VERSION;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(MPServerlessMiniAppCore.prototype, "ua", {
            get: function () {
                return PKGUA;
            },
            enumerable: true,
            configurable: true
        });
        MPServerlessMiniAppCore.prototype.createTransport = function (options) {
            _super.prototype.createTransport.call(this, options);
            this.transport.setAppSecret(options.clientSecret).setUA(this.ua);
        };
        return MPServerlessMiniAppCore;
    }(mpserverless_core_1.MPServerlessCore));
    exports.MPServerlessMiniAppCore = MPServerlessMiniAppCore;
});
