// alipay-serverless-sdk/src/payment/huabei/client.ts
// This file is auto-generated, don't edit it
import CloudKernel from 'mpserverless-sdk-core';
import { Readable } from 'stream';
import * as $tea from '@alicloud/tea-typescript';

export class HuabeiConfig extends $tea.Model {
  hbFqNum: string;
  hbFqSellerPercent: string;
  static names(): { [key: string]: string } {
    return {
      hbFqNum: 'hbFqNum',
      hbFqSellerPercent: 'hbFqSellerPercent',
    };
  }

  static types(): { [key: string]: any } {
    return {
      hbFqNum: 'string',
      hbFqSellerPercent: 'string',
    };
  }

  constructor(map?: { [key: string]: any }) {
    super(map);
  }
}

export class AlipayTradeCreateResponse extends $tea.Model {
  code: string;
  msg: string;
  subCode: string;
  subMsg: string;
  outTradeNo: string;
  tradeNo: string;
  static names(): { [key: string]: string } {
    return {
      code: 'code',
      msg: 'msg',
      subCode: 'subCode',
      subMsg: 'subMsg',
      outTradeNo: 'outTradeNo',
      tradeNo: 'tradeNo',
    };
  }

  static types(): { [key: string]: any } {
    return {
      code: 'string',
      msg: 'string',
      subCode: 'string',
      subMsg: 'string',
      outTradeNo: 'string',
      tradeNo: 'string',
    };
  }

  constructor(map?: { [key: string]: any }) {
    super(map);
  }
}


export default class Client {
  _kernel: CloudKernel;

  constructor(_kernel: CloudKernel) {
    this._kernel = _kernel;
  }


  async create(subject: string, outTradeNo: string, totalAmount: string, buyerId: string, extendParams: HuabeiConfig): Promise<AlipayTradeCreateResponse> {
    let params : {[key: string ]: any} = {
      subject: subject,
      out_trade_no: outTradeNo,
      total_amount: totalAmount,
      buyer_id: buyerId,
      extend_params: extendParams,
    };
    return $tea.cast<AlipayTradeCreateResponse>(await this._kernel.invoke("alipay.trade.create", params), new AlipayTradeCreateResponse({}));
  }

}
