// @alicloud/mpserverless-miniapp-core/dist/umd/codec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const mpserverless_core_1 = require("@alicloud/mpserverless-core");
const core_1 = tslib_1.__importDefault(require("crypto-js/core"));
const hmac_md5_1 = tslib_1.__importDefault(require("crypto-js/hmac-md5"));
const error_1 = require("./error");
class MPHTTPRequestEncoder extends mpserverless_core_1.HTTPRequestEncoder {
    constructor(endpoint, spaceId) {
        super(endpoint);
        this.spaceId = spaceId;
        this.prefix = mpserverless_core_1.PREFIX.CLIENT;
        this.serviceHeaders = {};
        this.setBodyField({
            spaceId,
        });
    }
    sign(clientSecret) {
        const { spaceId, method, params, token, userId } = this.body;
        const timestamp = Date.now();
        this.setBodyField({
            timestamp,
        });
        let signString = '';
        const signObject = {
            spaceId,
            timestamp,
            method,
            params: JSON.stringify(params),
            token,
            userId,
        };
        Object.keys(signObject).sort().forEach(key => {
            if (signObject[key]) {
                signString = `${signString}&${key}=${signObject[key]}`;
            }
        });
        signString = signString.slice(1);
        const sign = hmac_md5_1.default(signString, clientSecret).toString(core_1.default.enc.Hex);
        this.setServerlessHeaders({ sign });
    }
    encodeAsHTTPRequestObject(additionalObject) {
        if (this.body.params) {
            this.body.params = JSON.stringify(this.body.params);
        }
        return Object.assign({ url: this.url, data: this.body, method: this.method, headers: this.headers, dataType: 'json' }, additionalObject);
    }
    clone() {
        const encoder = new MPHTTPRequestEncoder(this.endpoint, this.spaceId);
        encoder.setBodyField(this.body);
        encoder.setBaseHeaders(this.baseHeaders);
        encoder.setServerlessHeaders(this.serverlessHeaders);
        return encoder;
    }
}
exports.MPHTTPRequestEncoder = MPHTTPRequestEncoder;
class MPHTTPResponseDecoder extends mpserverless_core_1.HTTPResponseDecoder {
    constructor() {
        super(...arguments);
        this.ERROR_CODES = [11, 12, 13, 14, 19, 20];
    }
    setStatusAndBody(status, body) {
        super.setStatusAndBody(status, body);
        if (this.ERROR_CODES.indexOf(status) >= 0) {
            this._error = error_1.bizError.InterfaceRequestError(`错误码 ${status.toString()}, 请参考对应开放平台文档排查问题`);
        }
    }
    decode(res) {
        this.setHeaders(res.headers || {});
        let body = res.data || res.body;
        if (typeof body === 'string') {
            body = JSON.parse(body);
        }
        if (body && body.data) {
            if (Number(body.data.affectedDocs) === body.data.affectedDocs) {
                body = Object.assign({}, body, Object.assign({}, body.data));
            }
            else if (Object.prototype.toString.call(body.data) === '[object Object]') {
                body.result = Object.assign({}, body.data);
            }
            else if (Object.prototype.toString.call(body.data) === '[object Array]') {
                [...body.result] = body.data;
            }
            else {
                body.result = body.data;
            }
            delete body.data;
        }
        if (/^request:fail+/.test(res.errMsg)) {
            this.setErrorMessage(res.errMsg);
            return super.decode();
        }
        const responseErrorCode = parseInt(res.error, 10);
        if (responseErrorCode) {
            this.setStatusAndBody(responseErrorCode, body);
            return super.decode();
        }
        const responseErrorMessage = res.err;
        if (responseErrorMessage) {
            this.setErrorMessage(responseErrorMessage);
            return super.decode();
        }
        if (res instanceof Error) {
            this.setErrorObject(res);
            return super.decode();
        }
        if (body && typeof body.error === 'object') {
            this.setErrorObject(body.error);
            return super.decode();
        }
        const responseStatusCode = parseInt(res.status || res.statusCode, 10);
        if (responseStatusCode) {
            this.setStatusAndBody(responseStatusCode, body);
            return super.decode();
        }
        this.setStatusAndBody(200, res);
        return super.decode();
    }
}
exports.MPHTTPResponseDecoder = MPHTTPResponseDecoder;
