// @alicloud/mpserverless-miniapp-core/dist/umd/mpserverless.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const mpserverless_core_1 = require("@alicloud/mpserverless-core");
const mp_proxy_service_1 = require("@alicloud/mp-proxy-service");
const mp_db_service_1 = require("@alicloud/mp-db-service");
const mp_user_service_1 = require("@alicloud/mp-user-service");
const mp_function_service_1 = require("@alicloud/mp-function-service");
const transport_1 = require("./transport");
const file_1 = require("./file");
class MPServerlessMiniAppCore extends mpserverless_core_1.MPServerlessCore {
    constructor(appGlobal, options) {
        super(Object.assign(Object.assign({}, options), { httpTransport: transport_1.MPHTTPTransport, options: appGlobal.logger }));
        mpserverless_core_1.assert(options.clientSecret, 'clientSecret is required');
        mpserverless_core_1.assert(options.appId, 'appId is required');
        this.user = new mp_user_service_1.UserService(this.transport, {
            getAuthCode: this.transport.wrapApi(appGlobal.getAuthCode),
            request: this.transport.wrapApi(appGlobal.request),
            appId: options.appId,
            appSecret: options.clientSecret,
            ua: this.ua,
        });
        this.transport.setUserService(this.user);
        this.transport.setRequest(this.transport.wrapApi(appGlobal.request));
        this.db = new mp_db_service_1.DbService(this.transport);
        this.file = new file_1.HighFileSevice(this.transport, {
            uploadFile: this.transport.wrapApi(appGlobal.uploadFile),
            getFileUploadInfo: this.transport.wrapApi(appGlobal.getFileUploadInfo),
        });
        this.function = new mp_function_service_1.FunctionService(this.transport);
        this.network = new mp_proxy_service_1.ProxyService(this.transport);
    }
    get version() {
        return VERSION;
    }
    get ua() {
        return PKGUA;
    }
    createTransport(options) {
        super.createTransport(options);
        this.transport.setAppSecret(options.clientSecret).setUA(this.ua);
    }
}
exports.MPServerlessMiniAppCore = MPServerlessMiniAppCore;
