// @alicloud/mpserverless-miniapp-core/dist/umd/transport.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const mpserverless_core_1 = require("@alicloud/mpserverless-core");
const codec_1 = require("./codec");
const error_1 = require("./error");
class MPHTTPTransport extends mpserverless_core_1.HTTPTransport {
    constructor(endpoint) {
        super(endpoint);
    }
    setUserService(userService) {
        this.userService = userService;
    }
    setRequest(request) {
        mpserverless_core_1.assert(request, '初始化时缺少 request 参数');
        this.httpRequest = request;
    }
    getEncoder(prefix) {
        return new codec_1.MPHTTPRequestEncoder(this.endpoint, this.spaceId);
    }
    request(encoder, retried = false) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const cloned = encoder.clone();
            const newEncoder = yield this.userService.setRequestToken(encoder, retried);
            if (this.ua) {
                newEncoder.setBaseHeaders({
                    'x-serverless-ua': this.ua,
                });
            }
            try {
                const encoded = newEncoder.encodeAsHTTPRequestObject({
                    timeout: this.timeout,
                    dataType: 'json',
                });
                this.logger.info('request encode data', encoded);
                const decoded = yield this.httpRequest(encoded);
                return decoded;
            }
            catch (e) {
                this.logger.error('request error', e);
                const isUnAuthorized = e.error.code === 'GATEWAY_INVALID_TOKEN'
                    || e.error.code === 'InvalidParameter.InvalidToken';
                if (isUnAuthorized) {
                    if (retried) {
                        throw new error_1.bizError.UnauthorizedError('授权 token 已过期，重新获取失败');
                    }
                    return yield this.request(cloned, true);
                }
                if (e.error) {
                    throw e.error;
                }
                throw e;
            }
        });
    }
    wrapApi(myMethod) {
        return (args) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            const res = yield myMethod(args);
            this.logger.info('completed request');
            this.logger.info(JSON.stringify(res, null, 2));
            const decoder = new codec_1.MPHTTPResponseDecoder();
            const response = decoder.decode(res);
            if (response.error) {
                throw response;
            }
            return response;
        });
    }
}
exports.MPHTTPTransport = MPHTTPTransport;
