// @alicloud/mpserverless-core/dist/esm/transport/index.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "tslib", "../constant", "../codec"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var tslib_1 = require("tslib");
    var constant_1 = require("../constant");
    var codec_1 = require("../codec");
    var HTTPTransport = (function () {
        function HTTPTransport(endpoint) {
            this.endpoint = endpoint;
            this.protocol = constant_1.TransportProtocol.HTTP;
        }
        HTTPTransport.prototype.getEncoder = function (prefix) {
            return new codec_1.HTTPRequestEncoder(this.endpoint, prefix);
        };
        HTTPTransport.prototype.setAppId = function (appId) {
            this.appId = appId;
        };
        HTTPTransport.prototype.setAppSecret = function (privateKey) {
            this.appSecret = privateKey;
            return this;
        };
        HTTPTransport.prototype.setUA = function (ua) {
            this.ua = ua;
            return this;
        };
        HTTPTransport.prototype.setLogger = function (logger) {
            this.logger = logger;
        };
        HTTPTransport.prototype.setTimeout = function (timeout) {
            if (timeout === void 0) { timeout = '5s'; }
            if (typeof timeout === 'string') {
                if (timeout.indexOf('ms') >= 0) {
                    this.timeout = parseInt(timeout, 10);
                    return;
                }
                if (timeout.indexOf('s') >= 0) {
                    this.timeout = parseInt(timeout, 10) * 1000;
                    return;
                }
            }
            else if (typeof timeout === 'number') {
                this.timeout = timeout;
                return;
            }
            this.timeout = 5000;
        };
        Object.defineProperty(HTTPTransport.prototype, "timeoutOption", {
            get: function () {
                return this.timeout;
            },
            enumerable: true,
            configurable: true
        });
        HTTPTransport.prototype.setSpaceId = function (spaceId) {
            this.spaceId = spaceId;
        };
        HTTPTransport.prototype.request = function (encoder) {
            return tslib_1.__awaiter(this, void 0, void 0, function () {
                return tslib_1.__generator(this, function (_a) {
                    return [2, (new codec_1.HTTPResponseDecoder()).decode()];
                });
            });
        };
        return HTTPTransport;
    }());
    exports.HTTPTransport = HTTPTransport;
});
