// alipay-serverless-sdk/dist/marketing/exchangevoucher/client.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var $tea = __importStar(require("@alicloud/tea-typescript"));
var BillDetail = /** @class */ (function (_super) {
    __extends(BillDetail, _super);
    function BillDetail(map) {
        return _super.call(this, map) || this;
    }
    BillDetail.names = function () {
        return {
            tradeNo: 'tradeNo',
            amount: 'amount',
            status: 'status',
            partnerId: 'partnerId',
            partnerName: 'partnerName',
            bizType: 'bizType',
            gmtCreate: 'gmtCreate',
        };
    };
    BillDetail.types = function () {
        return {
            tradeNo: 'string',
            amount: 'string',
            status: 'string',
            partnerId: 'string',
            partnerName: 'string',
            bizType: 'string',
            gmtCreate: 'string',
        };
    };
    return BillDetail;
}($tea.Model));
exports.BillDetail = BillDetail;
var AlipayMarktingExchangevoucherUseResponse = /** @class */ (function (_super) {
    __extends(AlipayMarktingExchangevoucherUseResponse, _super);
    function AlipayMarktingExchangevoucherUseResponse(map) {
        return _super.call(this, map) || this;
    }
    AlipayMarktingExchangevoucherUseResponse.names = function () {
        return {
            code: 'code',
            msg: 'msg',
            subCode: 'subCode',
            subMsg: 'subMsg',
            voucherId: 'voucherId',
        };
    };
    AlipayMarktingExchangevoucherUseResponse.types = function () {
        return {
            code: 'string',
            msg: 'string',
            subCode: 'string',
            subMsg: 'string',
            voucherId: 'string',
        };
    };
    return AlipayMarktingExchangevoucherUseResponse;
}($tea.Model));
exports.AlipayMarktingExchangevoucherUseResponse = AlipayMarktingExchangevoucherUseResponse;
var AlipayMarketingVoucherSendResponse = /** @class */ (function (_super) {
    __extends(AlipayMarketingVoucherSendResponse, _super);
    function AlipayMarketingVoucherSendResponse(map) {
        return _super.call(this, map) || this;
    }
    AlipayMarketingVoucherSendResponse.names = function () {
        return {
            code: 'code',
            msg: 'msg',
            subCode: 'subCode',
            subMsg: 'subMsg',
            voucherId: 'voucherId',
            userId: 'userId',
        };
    };
    AlipayMarketingVoucherSendResponse.types = function () {
        return {
            code: 'string',
            msg: 'string',
            subCode: 'string',
            subMsg: 'string',
            voucherId: 'string',
            userId: 'string',
        };
    };
    return AlipayMarketingVoucherSendResponse;
}($tea.Model));
exports.AlipayMarketingVoucherSendResponse = AlipayMarketingVoucherSendResponse;
var AlipayMarketingExchangevouherTemplateCreateResponse = /** @class */ (function (_super) {
    __extends(AlipayMarketingExchangevouherTemplateCreateResponse, _super);
    function AlipayMarketingExchangevouherTemplateCreateResponse(map) {
        return _super.call(this, map) || this;
    }
    AlipayMarketingExchangevouherTemplateCreateResponse.names = function () {
        return {
            code: 'code',
            msg: 'msg',
            subCode: 'subCode',
            subMsg: 'subMsg',
            templateId: 'templateId',
        };
    };
    AlipayMarketingExchangevouherTemplateCreateResponse.types = function () {
        return {
            code: 'string',
            msg: 'string',
            subCode: 'string',
            subMsg: 'string',
            templateId: 'string',
        };
    };
    return AlipayMarketingExchangevouherTemplateCreateResponse;
}($tea.Model));
exports.AlipayMarketingExchangevouherTemplateCreateResponse = AlipayMarketingExchangevouherTemplateCreateResponse;
var AlipayMarketingVoucherQueryResponse = /** @class */ (function (_super) {
    __extends(AlipayMarketingVoucherQueryResponse, _super);
    function AlipayMarketingVoucherQueryResponse(map) {
        return _super.call(this, map) || this;
    }
    AlipayMarketingVoucherQueryResponse.names = function () {
        return {
            code: 'code',
            msg: 'msg',
            subCode: 'subCode',
            subMsg: 'subMsg',
            name: 'name',
            voucherId: 'voucherId',
            userId: 'userId',
            templateId: 'templateId',
            gmtActive: 'gmtActive',
            gmtExpired: 'gmtExpired',
            totalAmount: 'totalAmount',
            availableAmount: 'availableAmount',
            status: 'status',
            gmtCreate: 'gmtCreate',
            extendInfo: 'extendInfo',
            billDetails: 'billDetails',
        };
    };
    AlipayMarketingVoucherQueryResponse.types = function () {
        return {
            code: 'string',
            msg: 'string',
            subCode: 'string',
            subMsg: 'string',
            name: 'string',
            voucherId: 'string',
            userId: 'string',
            templateId: 'string',
            gmtActive: 'string',
            gmtExpired: 'string',
            totalAmount: 'string',
            availableAmount: 'string',
            status: 'string',
            gmtCreate: 'string',
            extendInfo: 'string',
            billDetails: { 'type': 'array', 'itemType': BillDetail },
        };
    };
    return AlipayMarketingVoucherQueryResponse;
}($tea.Model));
exports.AlipayMarketingVoucherQueryResponse = AlipayMarketingVoucherQueryResponse;
var Client = /** @class */ (function () {
    function Client(_kernel) {
        this._kernel = _kernel;
    }
    Client.prototype.create = function (outBizNo, brandName, voucherName, quantity, publishStartTime, publishEndTime, validPeriod, availableTime, description, extensionInfo) {
        return __awaiter(this, void 0, void 0, function () {
            var params, _a, _b;
            return __generator(this, function (_c) {
                switch (_c.label) {
                    case 0:
                        params = {
                            out_biz_no: outBizNo,
                            voucher_type: "EXCHANGE_VOUCHER",
                            brand_name: brandName,
                            voucher_name: voucherName,
                            voucher_quantity: quantity,
                            publish_start_time: publishStartTime,
                            publish_end_time: publishEndTime,
                            voucher_valid_period: validPeriod,
                            voucher_available_time: availableTime,
                            voucher_description: description,
                            use_type: "H5",
                            extension_info: extensionInfo,
                        };
                        _b = (_a = $tea).cast;
                        return [4 /*yield*/, this._kernel.invoke("alipay.marketing.exchangevoucher.template.create", params)];
                    case 1: return [2 /*return*/, _b.apply(_a, [_c.sent(), new AlipayMarketingExchangevouherTemplateCreateResponse({})])];
                }
            });
        });
    };
    Client.prototype.send = function (templateId, userId, outBizNo) {
        return __awaiter(this, void 0, void 0, function () {
            var params, _a, _b;
            return __generator(this, function (_c) {
                switch (_c.label) {
                    case 0:
                        params = {
                            template_id: templateId,
                            user_id: userId,
                            out_biz_no: outBizNo,
                        };
                        _b = (_a = $tea).cast;
                        return [4 /*yield*/, this._kernel.invoke("alipay.marketing.voucher.send", params)];
                    case 1: return [2 /*return*/, _b.apply(_a, [_c.sent(), new AlipayMarketingVoucherSendResponse({})])];
                }
            });
        });
    };
    Client.prototype.use = function (voucherId, userId, outBizNo) {
        return __awaiter(this, void 0, void 0, function () {
            var params, _a, _b;
            return __generator(this, function (_c) {
                switch (_c.label) {
                    case 0:
                        params = {
                            voucher_id: voucherId,
                            user_id: userId,
                            out_biz_no: outBizNo,
                        };
                        _b = (_a = $tea).cast;
                        return [4 /*yield*/, this._kernel.invoke("alipay.marketing.exchangevoucher.use", params)];
                    case 1: return [2 /*return*/, _b.apply(_a, [_c.sent(), new AlipayMarktingExchangevoucherUseResponse({})])];
                }
            });
        });
    };
    Client.prototype.query = function (voucherId) {
        return __awaiter(this, void 0, void 0, function () {
            var params, _a, _b;
            return __generator(this, function (_c) {
                switch (_c.label) {
                    case 0:
                        params = {
                            voucher_id: voucherId,
                        };
                        _b = (_a = $tea).cast;
                        return [4 /*yield*/, this._kernel.invoke("alipay.marketing.voucher.query", params)];
                    case 1: return [2 /*return*/, _b.apply(_a, [_c.sent(), new AlipayMarketingVoucherQueryResponse({})])];
                }
            });
        });
    };
    return Client;
}());
exports.default = Client;
