// @alicloud/mpserverless-core/dist/umd/mpserverless.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const utility_1 = require("./utility");
class MPServerlessCore {
    constructor(options) {
        utility_1.assert(options, 'options is required');
        utility_1.assert(options.spaceId, 'spaceId is required');
        utility_1.assert(options.endpoint, 'endpoint is required');
        this._debug = false;
        this._logger = options.logger || utility_1.emptyLogger;
        this.options = options;
        this.createTransport(options);
    }
    setDebugFlag(flag) {
        this._debug = flag;
        this.transport.setLogger(this.logger);
    }
    get debug() {
        return this._debug;
    }
    get logger() {
        return this._debug ? this._logger : utility_1.emptyLogger;
    }
    createTransport(options) {
        const Klass = options.httpTransport;
        this.transport = new Klass(options.endpoint);
        this.transport.setAppId(options.appId);
        this.transport.setLogger(this.logger);
        this.transport.setSpaceId(options.spaceId);
        this.transport.setTimeout(options.timeout);
    }
}
exports.MPServerlessCore = MPServerlessCore;
