// @alicloud/mpserverless-core/dist/esm/codec/request.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "tslib", "../constant", "../utility"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var tslib_1 = require("tslib");
    var constant_1 = require("../constant");
    var utility_1 = require("../utility");
    var BaseEncoder = (function () {
        function BaseEncoder() {
        }
        return BaseEncoder;
    }());
    exports.BaseEncoder = BaseEncoder;
    var HTTPRequestEncoder = (function (_super) {
        tslib_1.__extends(HTTPRequestEncoder, _super);
        function HTTPRequestEncoder(endpoint, prefix) {
            var _this = _super.call(this) || this;
            _this.endpoint = endpoint;
            _this.body = {};
            _this.query = {};
            _this.method = constant_1.HTTPMethod.POST;
            _this.prefix = '';
            _this.baseHeaders = {};
            _this.serviceHeaders = {};
            _this.serverlessHeaders = {};
            if (prefix) {
                _this.prefix = prefix;
            }
            return _this;
        }
        HTTPRequestEncoder.prototype.encodeAsHTTPRequestObject = function (additionalObject) {
            return {
                url: '',
                data: {},
                method: constant_1.HTTPMethod.POST,
                headers: {},
            };
        };
        HTTPRequestEncoder.prototype.sign = function (secret) { };
        Object.defineProperty(HTTPRequestEncoder.prototype, "url", {
            get: function () {
                return [
                    this.endpoint + this.prefix,
                ]
                    .filter(function (p) { return !!p; })
                    .join('?');
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(HTTPRequestEncoder.prototype, "headers", {
            get: function () {
                var _this = this;
                var normalizedHeaders = Object
                    .keys(this.serverlessHeaders)
                    .reduce(function (accu, prop) {
                    var key = "" + constant_1.SERVERLESS_HEADER_PREFIX + utility_1.camelToLisp(prop);
                    accu[key] = _this.serverlessHeaders[prop];
                    return accu;
                }, {});
                return tslib_1.__assign(tslib_1.__assign({}, this.baseHeaders), normalizedHeaders);
            },
            enumerable: true,
            configurable: true
        });
        HTTPRequestEncoder.prototype.setBodyField = function (fields) {
            this.body = Object.assign({}, this.body, tslib_1.__assign({}, fields));
            return this;
        };
        HTTPRequestEncoder.prototype.setUserId = function (userId) {
            this.setBodyField({
                userId: userId,
            });
            return this;
        };
        HTTPRequestEncoder.prototype.setBaseHeaders = function (headers) {
            if (headers === void 0) { headers = {}; }
            this.baseHeaders = tslib_1.__assign(tslib_1.__assign({}, this.baseHeaders), Object.keys(headers).reduce(function (accu, key) {
                if (headers[key]) {
                    accu[key] = headers[key].toString();
                }
                return accu;
            }, {}));
            return this;
        };
        HTTPRequestEncoder.prototype.setServerlessHeaders = function (headers) {
            if (headers === void 0) { headers = {}; }
            this.serverlessHeaders = tslib_1.__assign(tslib_1.__assign({}, this.serverlessHeaders), Object.keys(headers).reduce(function (accu, key) {
                if (headers[key]) {
                    accu[key] = headers[key].toString();
                }
                return accu;
            }, {}));
            return this;
        };
        HTTPRequestEncoder.prototype.encode = function () {
            var _ = [];
            for (var _i = 0; _i < arguments.length; _i++) {
                _[_i] = arguments[_i];
            }
            if (this.body.params) {
                this.body.params = JSON.stringify(this.body.params);
            }
            return {
                url: this.url,
                data: this.body,
                method: this.method,
                headers: this.headers,
            };
        };
        return HTTPRequestEncoder;
    }(BaseEncoder));
    exports.HTTPRequestEncoder = HTTPRequestEncoder;
});
