// @alicloud/mpserverless-core/dist/esm/codec/response.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "tslib"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var tslib_1 = require("tslib");
    var BaseDecoder = (function () {
        function BaseDecoder() {
        }
        return BaseDecoder;
    }());
    exports.BaseDecoder = BaseDecoder;
    var HTTPResponseDecoder = (function (_super) {
        tslib_1.__extends(HTTPResponseDecoder, _super);
        function HTTPResponseDecoder() {
            var _this = _super !== null && _super.apply(this, arguments) || this;
            _this._body = {};
            _this._headers = {};
            return _this;
        }
        HTTPResponseDecoder.prototype.setHeaders = function (headers) {
            this._headers = tslib_1.__assign(tslib_1.__assign({}, this._headers), headers);
        };
        HTTPResponseDecoder.prototype.setStatusAndBody = function (status, body) {
            this._status = status;
            this._body = body;
        };
        HTTPResponseDecoder.prototype.setErrorMessage = function (message) {
            this._error = new Error(message);
        };
        HTTPResponseDecoder.prototype.setErrorObject = function (error) {
            this._error = error;
        };
        HTTPResponseDecoder.prototype.decode = function () {
            var _ = [];
            for (var _i = 0; _i < arguments.length; _i++) {
                _[_i] = arguments[_i];
            }
            return {
                body: this._body || {},
                error: this._error,
                status: this._status,
                headers: this._headers,
            };
        };
        return HTTPResponseDecoder;
    }(BaseDecoder));
    exports.HTTPResponseDecoder = HTTPResponseDecoder;
});
