// alipay-serverless-sdk/src/security/textRisk/client.ts
// This file is auto-generated, don't edit it
import CloudKernel from 'mpserverless-sdk-core';
import { Readable } from 'stream';
import * as $tea from '@alicloud/tea-typescript';

export class AlipaySecurityRiskContentDetectResponse extends $tea.Model {
  code: string;
  msg: string;
  subCode: string;
  subMsg: string;
  action: string;
  keywords: string[];
  uniqueId: string;
  static names(): { [key: string]: string } {
    return {
      code: 'code',
      msg: 'msg',
      subCode: 'subCode',
      subMsg: 'subMsg',
      action: 'action',
      keywords: 'keywords',
      uniqueId: 'uniqueId',
    };
  }

  static types(): { [key: string]: any } {
    return {
      code: 'string',
      msg: 'string',
      subCode: 'string',
      subMsg: 'string',
      action: 'string',
      keywords: { 'type': 'array', 'itemType': 'string' },
      uniqueId: 'string',
    };
  }

  constructor(map?: { [key: string]: any }) {
    super(map);
  }
}


export default class Client {
  _kernel: CloudKernel;

  constructor(_kernel: CloudKernel) {
    this._kernel = _kernel;
  }


  async detect(content: string): Promise<AlipaySecurityRiskContentDetectResponse> {
    let params : {[key: string ]: any} = {
      content: content,
    };
    return $tea.cast<AlipaySecurityRiskContentDetectResponse>(await this._kernel.invoke("alipay.security.risk.content.detect", params), new AlipaySecurityRiskContentDetectResponse({}));
  }

}
