// @alicloud/mpserverless-core/dist/umd/codec/request.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const constant_1 = require("../constant");
const utility_1 = require("../utility");
class BaseEncoder {
}
exports.BaseEncoder = BaseEncoder;
class HTTPRequestEncoder extends BaseEncoder {
    constructor(endpoint, prefix) {
        super();
        this.endpoint = endpoint;
        this.body = {};
        this.query = {};
        this.method = constant_1.HTTPMethod.POST;
        this.prefix = '';
        this.baseHeaders = {};
        this.serviceHeaders = {};
        this.serverlessHeaders = {};
        if (prefix) {
            this.prefix = prefix;
        }
    }
    encodeAsHTTPRequestObject(additionalObject) {
        return {
            url: '',
            data: {},
            method: constant_1.HTTPMethod.POST,
            headers: {},
        };
    }
    sign(secret) { }
    get url() {
        return [
            this.endpoint + this.prefix,
        ]
            .filter(p => !!p)
            .join('?');
    }
    get headers() {
        const normalizedHeaders = Object
            .keys(this.serverlessHeaders)
            .reduce((accu, prop) => {
            const key = `${constant_1.SERVERLESS_HEADER_PREFIX}${utility_1.camelToLisp(prop)}`;
            accu[key] = this.serverlessHeaders[prop];
            return accu;
        }, {});
        return Object.assign(Object.assign({}, this.baseHeaders), normalizedHeaders);
    }
    setBodyField(fields) {
        this.body = Object.assign({}, this.body, Object.assign({}, fields));
        return this;
    }
    setUserId(userId) {
        this.setBodyField({
            userId,
        });
        return this;
    }
    setBaseHeaders(headers = {}) {
        this.baseHeaders = Object.assign(Object.assign({}, this.baseHeaders), Object.keys(headers).reduce((accu, key) => {
            if (headers[key]) {
                accu[key] = headers[key].toString();
            }
            return accu;
        }, {}));
        return this;
    }
    setServerlessHeaders(headers = {}) {
        this.serverlessHeaders = Object.assign(Object.assign({}, this.serverlessHeaders), Object.keys(headers).reduce((accu, key) => {
            if (headers[key]) {
                accu[key] = headers[key].toString();
            }
            return accu;
        }, {}));
        return this;
    }
    encode(..._) {
        if (this.body.params) {
            this.body.params = JSON.stringify(this.body.params);
        }
        return {
            url: this.url,
            data: this.body,
            method: this.method,
            headers: this.headers,
        };
    }
}
exports.HTTPRequestEncoder = HTTPRequestEncoder;
