// @alicloud/mpserverless-alipay-sdk/dist/esm/mpserverless.js
import { MPServerlessMiniAppCore, } from '@alicloud/mpserverless-miniapp-core';
export class MPServerless extends MPServerlessMiniAppCore {
    constructor(options) {
        super({
            request: (args) => {
                return this.wrap(my.request)(args);
            },
            getAuthCode: (args) => {
                return this.wrap(my.getAuthCode)(args);
            },
            uploadFile: (args) => {
                return this.wrap(my.uploadFile)(args);
            },
            getFileUploadInfo: (options) => {
                let fileSize = 0;
                let fileExtension = '';
                return this.wrap(my.getFileInfo)({
                    apFilePath: options.filePath,
                }).then(fileInfoRes => {
                    if (fileInfoRes.size) {
                        fileSize = fileInfoRes.size;
                    }
                    return this.wrap(my.getImageInfo)({
                        src: options.filePath,
                    });
                }).then(imageInfoRes => {
                    if (imageInfoRes.type) {
                        fileExtension = imageInfoRes.type;
                    }
                    return {
                        fileSize,
                        fileExtension,
                    };
                });
            },
        }, options);
    }
    wrap(myMethod) {
        return args => {
            return new Promise(resolve => {
                myMethod(Object.assign(args, {
                    complete: (res = {}) => {
                        return resolve(res);
                    },
                }));
            });
        };
    }
}
