// alipay-serverless-sdk/src/base/oauth/client.ts
// This file is auto-generated, don't edit it
import CloudKernel from 'mpserverless-sdk-core';
import { Readable } from 'stream';
import * as $tea from '@alicloud/tea-typescript';

export class AlipaySystemOauthTokenResponse extends $tea.Model {
  code: string;
  msg: string;
  subCode: string;
  subMsg: string;
  userId: string;
  accessToken: string;
  expiresIn: number;
  refreshToken: string;
  reExpiresIn: number;
  static names(): { [key: string]: string } {
    return {
      code: 'code',
      msg: 'msg',
      subCode: 'subCode',
      subMsg: 'subMsg',
      userId: 'userId',
      accessToken: 'accessToken',
      expiresIn: 'expiresIn',
      refreshToken: 'refreshToken',
      reExpiresIn: 'reExpiresIn',
    };
  }

  static types(): { [key: string]: any } {
    return {
      code: 'string',
      msg: 'string',
      subCode: 'string',
      subMsg: 'string',
      userId: 'string',
      accessToken: 'string',
      expiresIn: 'number',
      refreshToken: 'string',
      reExpiresIn: 'number',
    };
  }

  constructor(map?: { [key: string]: any }) {
    super(map);
  }
}


export default class Client {
  _kernel: CloudKernel;

  constructor(_kernel: CloudKernel) {
    this._kernel = _kernel;
  }


  async getToken(code: string): Promise<AlipaySystemOauthTokenResponse> {
    let params : {[key: string ]: any} = {
      grant_type: "authorization_code",
      code: code,
    };
    return $tea.cast<AlipaySystemOauthTokenResponse>(await this._kernel.invoke("alipay.system.oauth.token", params), new AlipaySystemOauthTokenResponse({}));
  }

  async refreshToken(refreshToken: string): Promise<AlipaySystemOauthTokenResponse> {
    let params : {[key: string ]: any} = {
      grant_type: "refresh_token",
      refresh_token: refreshToken,
    };
    return $tea.cast<AlipaySystemOauthTokenResponse>(await this._kernel.invoke("alipay.system.oauth.token", params), new AlipaySystemOauthTokenResponse({}));
  }

}
