// alipay-serverless-sdk/dist/aliyun/shortMessage/client.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var $tea = __importStar(require("@alicloud/tea-typescript"));
var SendSmsResponse = /** @class */ (function (_super) {
    __extends(SendSmsResponse, _super);
    function SendSmsResponse(map) {
        return _super.call(this, map) || this;
    }
    SendSmsResponse.names = function () {
        return {
            requestId: 'RequestId',
            code: 'Code',
            message: 'Message',
            bizId: 'BizId',
        };
    };
    SendSmsResponse.types = function () {
        return {
            requestId: 'string',
            code: 'string',
            message: 'string',
            bizId: 'string',
        };
    };
    return SendSmsResponse;
}($tea.Model));
exports.SendSmsResponse = SendSmsResponse;
var SendBatchSmsResponse = /** @class */ (function (_super) {
    __extends(SendBatchSmsResponse, _super);
    function SendBatchSmsResponse(map) {
        return _super.call(this, map) || this;
    }
    SendBatchSmsResponse.names = function () {
        return {
            requestId: 'RequestId',
            code: 'Code',
            message: 'Message',
            bizId: 'BizId',
        };
    };
    SendBatchSmsResponse.types = function () {
        return {
            requestId: 'string',
            code: 'string',
            message: 'string',
            bizId: 'string',
        };
    };
    return SendBatchSmsResponse;
}($tea.Model));
exports.SendBatchSmsResponse = SendBatchSmsResponse;
var QuerySmsResponse = /** @class */ (function (_super) {
    __extends(QuerySmsResponse, _super);
    function QuerySmsResponse(map) {
        return _super.call(this, map) || this;
    }
    QuerySmsResponse.names = function () {
        return {
            requestId: 'RequestId',
            code: 'Code',
            message: 'Message',
            smsSendDetailDTOs: 'SmsSendDetailDTOs',
            totalCount: 'TotalCount',
        };
    };
    QuerySmsResponse.types = function () {
        return {
            requestId: 'string',
            code: 'string',
            message: 'string',
            smsSendDetailDTOs: SendDetailInfo,
            totalCount: 'number',
        };
    };
    return QuerySmsResponse;
}($tea.Model));
exports.QuerySmsResponse = QuerySmsResponse;
var SendDetailInfo = /** @class */ (function (_super) {
    __extends(SendDetailInfo, _super);
    function SendDetailInfo(map) {
        return _super.call(this, map) || this;
    }
    SendDetailInfo.names = function () {
        return {
            smsSendDetailDTO: 'SmsSendDetailDTO',
        };
    };
    SendDetailInfo.types = function () {
        return {
            smsSendDetailDTO: { 'type': 'array', 'itemType': 'object' },
        };
    };
    return SendDetailInfo;
}($tea.Model));
exports.SendDetailInfo = SendDetailInfo;
var Client = /** @class */ (function () {
    function Client(_kernel) {
        this._kernel = _kernel;
    }
    Client.prototype.send = function (phoneNumbers, signName, templateCode, templateParam) {
        return __awaiter(this, void 0, void 0, function () {
            var params, _a, _b;
            return __generator(this, function (_c) {
                switch (_c.label) {
                    case 0:
                        params = {
                            PhoneNumbers: phoneNumbers,
                            SignName: signName,
                            TemplateCode: templateCode,
                            TemplateParam: templateParam,
                        };
                        _b = (_a = $tea).cast;
                        return [4 /*yield*/, this._kernel.invokePre("alicloud-dysms", "SendSms", params)];
                    case 1: return [2 /*return*/, _b.apply(_a, [_c.sent(), new SendSmsResponse({})])];
                }
            });
        });
    };
    Client.prototype.batchSend = function (phoneNumberJson, signNameJson, templateCode, templateParamJson) {
        return __awaiter(this, void 0, void 0, function () {
            var params, _a, _b;
            return __generator(this, function (_c) {
                switch (_c.label) {
                    case 0:
                        params = {
                            PhoneNumberJson: phoneNumberJson,
                            SignNameJson: signNameJson,
                            TemplateCode: templateCode,
                            TemplateParamJson: templateParamJson,
                        };
                        _b = (_a = $tea).cast;
                        return [4 /*yield*/, this._kernel.invokePre("alicloud-dysms", "SendBatchSms", params)];
                    case 1: return [2 /*return*/, _b.apply(_a, [_c.sent(), new SendBatchSmsResponse({})])];
                }
            });
        });
    };
    Client.prototype.query = function (phoneNumber, sendDate, pageSize, currentPage) {
        return __awaiter(this, void 0, void 0, function () {
            var params, _a, _b;
            return __generator(this, function (_c) {
                switch (_c.label) {
                    case 0:
                        params = {
                            PhoneNumber: phoneNumber,
                            SendDate: sendDate,
                            PageSize: pageSize,
                            CurrentPage: currentPage,
                        };
                        _b = (_a = $tea).cast;
                        return [4 /*yield*/, this._kernel.invokePre("alicloud-dysms", "QuerySendDetails", params)];
                    case 1: return [2 /*return*/, _b.apply(_a, [_c.sent(), new QuerySmsResponse({})])];
                }
            });
        });
    };
    Client.prototype.preciseQuery = function (phoneNumber, sendDate, pageSize, currentPage, bizId) {
        return __awaiter(this, void 0, void 0, function () {
            var params, _a, _b;
            return __generator(this, function (_c) {
                switch (_c.label) {
                    case 0:
                        params = {
                            PhoneNumber: phoneNumber,
                            SendDate: sendDate,
                            PageSize: pageSize,
                            CurrentPage: currentPage,
                            BizId: bizId,
                        };
                        _b = (_a = $tea).cast;
                        return [4 /*yield*/, this._kernel.invokePre("alicloud-dysms", "QuerySendDetails", params)];
                    case 1: return [2 /*return*/, _b.apply(_a, [_c.sent(), new QuerySmsResponse({})])];
                }
            });
        });
    };
    return Client;
}());
exports.default = Client;
