// @alicloud/mpserverless-core/dist/esm/constant/index.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var HTTPMethod;
    (function (HTTPMethod) {
        HTTPMethod["GET"] = "GET";
        HTTPMethod["DEL"] = "DELETE";
        HTTPMethod["POST"] = "POST";
    })(HTTPMethod = exports.HTTPMethod || (exports.HTTPMethod = {}));
    var TransportProtocol;
    (function (TransportProtocol) {
        TransportProtocol["HTTP"] = "HTTP";
    })(TransportProtocol = exports.TransportProtocol || (exports.TransportProtocol = {}));
    exports.SERVERLESS_HEADER_PREFIX = 'x-serverless-';
    var PREFIX;
    (function (PREFIX) {
        PREFIX["CLIENT"] = "/client";
        PREFIX["SERVER"] = "/server";
        PREFIX["ANTCLOUD"] = "/antcloud";
        PREFIX["ANTOPENANTCLOUD"] = "/antopen/antcloud";
    })(PREFIX = exports.PREFIX || (exports.PREFIX = {}));
});
