// @alicloud/mpserverless-core/dist/esm/utility/error.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "tslib"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var tslib_1 = require("tslib");
    function MPServerlessErrorClass(errorConfig, options) {
        if (options === void 0) { options = {
            code: 'MPServelessError',
            message: 'mpserverless error',
        }; }
        var error = {};
        var ErrorClass = (function (_super) {
            tslib_1.__extends(ErrorClass, _super);
            function ErrorClass(err, customMessage) {
                var _this = _super.call(this) || this;
                _this.code = options.code;
                _this.message = options.message;
                if (Error.captureStackTrace) {
                    Error.captureStackTrace(_this, ErrorClass);
                }
                if (customMessage) {
                    _this.message = _this.message + ": " + customMessage;
                }
                if (err instanceof Error) {
                    _this.message = _this.message + " << " + err.message;
                }
                else if (typeof err === 'string') {
                    _this.message = _this.message + ": " + err;
                }
                return _this;
            }
            return ErrorClass;
        }(Error));
        Object.keys(errorConfig).forEach(function (e) {
            var EKlass = (function (_super) {
                tslib_1.__extends(EKlass, _super);
                function EKlass(err, customMessage) {
                    var _this = _super.call(this, err) || this;
                    if (Error.captureStackTrace) {
                        Error.captureStackTrace(_this, EKlass);
                    }
                    _this.message = customMessage ? errorConfig[e].message + ": " + customMessage : errorConfig[e].message;
                    _this.code = errorConfig[e].code;
                    if (err instanceof Error) {
                        _this.message = _this.message + " << " + err.message;
                    }
                    else if (typeof err === 'string') {
                        _this.message = _this.message + ": " + err;
                    }
                    return _this;
                }
                EKlass.code = errorConfig[e].code;
                return EKlass;
            }(ErrorClass));
            error[e] = EKlass;
        });
        return {
            ErrorClass: ErrorClass,
            error: error,
        };
    }
    exports.MPServerlessErrorClass = MPServerlessErrorClass;
});
