// @alicloud/mpserverless-core/dist/esm/utility/index.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "tslib", "./error", "../error"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var tslib_1 = require("tslib");
    tslib_1.__exportStar(require("./error"), exports);
    var error_1 = require("../error");
    function assert(expr, message) {
        if (!expr) {
            throw new error_1.bizError.ValidationError(message);
        }
    }
    exports.assert = assert;
    function camelToLisp(key) {
        return key.replace(/[A-Z]/g, function (match) {
            return "-" + match.toLowerCase();
        });
    }
    exports.camelToLisp = camelToLisp;
    exports.emptyLogger = {
        log: function () { return; },
        info: function () { return; },
        warn: function () { return; },
        error: function () { return; },
        debug: function () { return; },
    };
});
