// alipay-serverless-sdk/dist/payment/faceToFace/client.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var $tea = __importStar(require("@alicloud/tea-typescript"));
var TradeFundBill = /** @class */ (function (_super) {
    __extends(TradeFundBill, _super);
    function TradeFundBill(map) {
        return _super.call(this, map) || this;
    }
    TradeFundBill.names = function () {
        return {
            fundChannel: 'fundChannel',
            bankCode: 'bankCode',
            amount: 'amount',
            realAmount: 'realAmount',
        };
    };
    TradeFundBill.types = function () {
        return {
            fundChannel: 'string',
            bankCode: 'string',
            amount: 'string',
            realAmount: 'string',
        };
    };
    return TradeFundBill;
}($tea.Model));
exports.TradeFundBill = TradeFundBill;
var VoucherDetail = /** @class */ (function (_super) {
    __extends(VoucherDetail, _super);
    function VoucherDetail(map) {
        return _super.call(this, map) || this;
    }
    VoucherDetail.names = function () {
        return {
            id: 'id',
            name: 'name',
            type: 'type',
            amount: 'amount',
            merchantContribute: 'merchantContribute',
            otherContribute: 'otherContribute',
            memo: 'memo',
            templateId: 'templateId',
            purchaseBuyerContribute: 'purchaseBuyerContribute',
            purchaseMerchantContribute: 'purchaseMerchantContribute',
            purchaseAntContribute: 'purchaseAntContribute',
        };
    };
    VoucherDetail.types = function () {
        return {
            id: 'string',
            name: 'string',
            type: 'string',
            amount: 'string',
            merchantContribute: 'string',
            otherContribute: 'string',
            memo: 'string',
            templateId: 'string',
            purchaseBuyerContribute: 'string',
            purchaseMerchantContribute: 'string',
            purchaseAntContribute: 'string',
        };
    };
    return VoucherDetail;
}($tea.Model));
exports.VoucherDetail = VoucherDetail;
var AlipayTradePayResponse = /** @class */ (function (_super) {
    __extends(AlipayTradePayResponse, _super);
    function AlipayTradePayResponse(map) {
        return _super.call(this, map) || this;
    }
    AlipayTradePayResponse.names = function () {
        return {
            code: 'code',
            msg: 'msg',
            subCode: 'subCode',
            subMsg: 'subMsg',
            tradeNo: 'tradeNo',
            outTradeNo: 'outTradeNo',
            buyerLogonId: 'buyerLogonId',
            settleAmount: 'settleAmount',
            payCurrency: 'payCurrency',
            payAmount: 'payAmount',
            settleTransRate: 'settleTransRate',
            transPayRate: 'transPayRate',
            totalAmount: 'totalAmount',
            transCurrency: 'transCurrency',
            settleCurrency: 'settleCurrency',
            receiptAmount: 'receiptAmount',
            buyerPayAmount: 'buyerPayAmount',
            pointAmount: 'pointAmount',
            invoiceAmount: 'invoiceAmount',
            gmtPayment: 'gmtPayment',
            fundBillList: 'fundBillList',
            cardBalance: 'cardBalance',
            storeName: 'storeName',
            buyerUserId: 'buyerUserId',
            discountGoodsDetail: 'discountGoodsDetail',
            voucherDetailList: 'voucherDetailList',
            advanceAmount: 'advanceAmount',
            authTradePayMode: 'authTradePayMode',
            chargeAmount: 'chargeAmount',
            chargeFlags: 'chargeFlags',
            settlementId: 'settlementId',
            businessParams: 'businessParams',
            buyerUserType: 'buyerUserType',
            mdiscountAmount: 'mdiscountAmount',
            discountAmount: 'discountAmount',
            buyerUserName: 'buyerUserName',
        };
    };
    AlipayTradePayResponse.types = function () {
        return {
            code: 'string',
            msg: 'string',
            subCode: 'string',
            subMsg: 'string',
            tradeNo: 'string',
            outTradeNo: 'string',
            buyerLogonId: 'string',
            settleAmount: 'string',
            payCurrency: 'string',
            payAmount: 'string',
            settleTransRate: 'string',
            transPayRate: 'string',
            totalAmount: 'string',
            transCurrency: 'string',
            settleCurrency: 'string',
            receiptAmount: 'string',
            buyerPayAmount: 'string',
            pointAmount: 'string',
            invoiceAmount: 'string',
            gmtPayment: 'string',
            fundBillList: { 'type': 'array', 'itemType': TradeFundBill },
            cardBalance: 'string',
            storeName: 'string',
            buyerUserId: 'string',
            discountGoodsDetail: 'string',
            voucherDetailList: { 'type': 'array', 'itemType': VoucherDetail },
            advanceAmount: 'string',
            authTradePayMode: 'string',
            chargeAmount: 'string',
            chargeFlags: 'string',
            settlementId: 'string',
            businessParams: 'string',
            buyerUserType: 'string',
            mdiscountAmount: 'string',
            discountAmount: 'string',
            buyerUserName: 'string',
        };
    };
    return AlipayTradePayResponse;
}($tea.Model));
exports.AlipayTradePayResponse = AlipayTradePayResponse;
var Client = /** @class */ (function () {
    function Client(_kernel) {
        this._kernel = _kernel;
    }
    Client.prototype.pay = function (subject, outTradeNo, totalAmount, authCode) {
        return __awaiter(this, void 0, void 0, function () {
            var params, _a, _b;
            return __generator(this, function (_c) {
                switch (_c.label) {
                    case 0:
                        params = {
                            subject: subject,
                            out_trade_no: outTradeNo,
                            total_amount: totalAmount,
                            auth_code: authCode,
                            scene: "bar_code",
                        };
                        _b = (_a = $tea).cast;
                        return [4 /*yield*/, this._kernel.invoke("alipay.trade.pay", params)];
                    case 1: return [2 /*return*/, _b.apply(_a, [_c.sent(), new AlipayTradePayResponse({})])];
                }
            });
        });
    };
    return Client;
}());
exports.default = Client;
